### DISCUSSION TITLE
Why does appdomain.currentdomain.setupinformation.targetframeworkname always be 4.5.1

### DISCUSSION THREAD

**Discussion by darkterror-jl** [xunit/xunit#2340] [Question] [answered]
Although the targeting framework set to 4.7.1(net471) or 4.8(net48)

What impacts: 
The attribute `ServicePointManager.SecurityProtocol` will be `SSL3|TLS` in `4.5.1`. but what I needed is `TLS1.2` which are supported in net `4.7.*` or later

**Comment by bradwilson** [comment, marked as answer]
In v2: this is a function of the requested .NET Framework used by the runner, and not the requested .NET Framework for your unit test project, because this decision is made at the time that the process is launched. (FWIW, this is one of the many reasons why we are [moving to make unit tests applications instead of libraries](https://github.com/xunit/xunit/issues/2331) in v3).

If you're using our console runner, we ship versions compiled against every version of .NET Framework between 4.5.2 and 4.7.2 (as these were the current versions of .NET Framework when we shipped xUnit.net 2.4.1). You can run the console runner that matches your desired version of .NET Framework, and that should bring along all the features (like TLS 1.2) that are available for that version of .NET.
